src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -189,14 +209,14 @@ protected override void StartPipelineExecution()
                         {
                             // if this a nested pipeline we are already in the appropriate thread so we just execute the pipeline here
                             SetupInvokeThread(Thread.CurrentThread, true);
-                            this.InvokeThreadProc();
+                            InvokeThreadProc();
```

Comment:
Why not using `invokeThreadProcDelegate` here?
By looking at the code, calling `InvokeThreadProc` here would honor the `FlowImpersonationPolicy ` as well in Windows PowerShell.